waku/node/peer_manager/peer_manager.nim

```diff
@@ -121,6 +121,7 @@ proc addPeer*(pm: PeerManager, remotePeerInfo: RemotePeerInfo, origin = UnknownO
   discard remotePeerInfo.peerId.extractPublicKey(publicKey)
 
   if pm.peerStore[AddressBook][remotePeerInfo.peerId] == remotePeerInfo.addrs and
+     not ($remotePeerInfo.addrs).contains("127.0.0.1") and
```

Comment:
Can you explain why this is needed, please? I cannot wrap my head around it, sorry:D

Reply:
Yes! So we were having issues with the peer exchange tests in which we weren't able to add peers when running locally after the initial change we made. `pm.peerStore[AddressBook][remotePeerInfo.peerId]` equals `127.0.0.1` in local runs, and `remotePeerInfo.addrs` used to equal `0.0.0.0` but after our fix to the Identify message, it equals `127.0.0.1`.

Because this if statement was always yielding true, it considered the IP `127.0.0.1` as duplicate, returned early from the addPeer procedure and we weren't able to add local peers. So we added a special case to bypass that check when it comes to localhost.

Reply:
But that is strange, isn't it? We are also checking the publicKey, so that would mean that not only the address is the loopback interface, but also the pubKey is the same - IOW this check should be fals if you have multiple local node peers as long as they have different peerId, so the explicit check for 127.0.0.1 should not be necessary. Or I am still missing something:D

Can you share which tests were failing?

Reply:
Now that I look further into it I agree that there's something weird in there.
Maybe the public key check is reduntant/wrong? If `pm.peerStore[KeyBook][remotePeerInfo.peerId]` has to have the value of the public key for a peerId, and `publicKey` is a variable with the public key of the peer we are checking - aren't they supposed to be always the same?

The test that fails without that line is `nwaku/tests/test_waku_peer_exchange test 'retrieve and provide peer exchange peers from discv5'`

Reply:
> Weird, I commented out this line and ran the test you mentioned locall and everything passed:)

That's really weird, without that line I couldn't get the test to pass neither locally nor in the CI. I'll DM you to have a look :)